File: app/src/main/java/com/example/myapplicationmo1/LoginActivity.java
package com.example.myapplicationmo1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import static com.example.myapplicationmo1.PrefConstant.IS_LOGGED_IN;

public class LoginActivity extends AppCompatActivity {

    EditText editTextFullName,editTextUserName;
    Button buttonLogin;
    SharedPreferences sharedPreferences;
    SharedPreferences.Editor editor;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        editTextFullName = findViewById(R.id.editTextFullName);
        editTextUserName = findViewById(R.id.editTextUserName);
        buttonLogin = findViewById(R.id.buttonLogin);
        setupSharedPreferences();

       View.OnClickListener clickAction = new View.OnClickListener(){
          @Override
          public void onClick(View view){

              String fullName = editTextFullName.getText().toString();
              String UserName = editTextUserName.getText().toString();
              //full name + username shouldn't be empty
              if(!TextUtils.isEmpty(fullName) && !TextUtils.isEmpty(UserName)) {

                  Intent intent = new Intent(LoginActivity.this, MyNotesActivity.class);
                  intent.putExtra(AppConstant.Full_Name, fullName);
                  startActivity(intent);

                  saveLoginStatus();
                  saveFullName(fullName);
              } else{
                  Toast.makeText(LoginActivity.this,"FullName and UserName both are mendatory",Toast.LENGTH_SHORT).show();
              }
          }
       };
           buttonLogin.setOnClickListener(clickAction);
    }

    private void saveFullName(String fullName) {
        editor = sharedPreferences.edit();
        editor.putString(PrefConstant.FULL_NAME,fullName);
        editor.apply();
    }

    private void saveLoginStatus() {
        editor = sharedPreferences.edit();
        editor.putBoolean(IS_LOGGED_IN,true);
        editor.apply();
    }

    private void setupSharedPreferences() {
        sharedPreferences = getSharedPreferences(PrefConstant.SHARED_PREFERENCE_NAME,MODE_PRIVATE);
    }
}

File: app/src/main/java/com/example/myapplicationmo1/view/LoginActivity.kt
package com.example.myapplicationmo1.view

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.myapplicationmo1.utils.AppConstant
import com.example.myapplicationmo1.utils.PrefConstant
import com.example.myapplicationmo1.R
import com.example.myapplicationmo1.utils.StoreSession

class LoginActivity : AppCompatActivity() {
    lateinit var editTextFullName:EditText
    lateinit var editTextUserName: EditText
    lateinit var buttonLogin: Button
    lateinit var sharedPreferences: SharedPreferences
    lateinit var editer: SharedPreferences.Editor
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        bindViews()
        setupSharedPreference()
    }

    private fun setupSharedPreference() {
       StoreSession.init(this)
    }

    private fun bindViews() {
        editTextFullName = findViewById(R.id.editTextFullName)
        editTextUserName = findViewById(R.id.editTextUserName)
        buttonLogin = findViewById(R.id.buttonLogin)
        val clickAction = object :  View.OnClickListener{
            override fun onClick(v: View?) {
                val fullName = editTextFullName.text.toString()
                val userName = editTextUserName.text.toString()
                if(fullName.isNotEmpty() && userName.isNotEmpty()){
                   val intent = Intent(this@LoginActivity, MyNotesActivity::class.java)
                    intent.putExtra(AppConstant.Full_Name,fullName)
                    startActivity(intent)
                    saveFullName(fullName)
                    saveLoginState()
                }else{
                      val toast = Toast.makeText(this@LoginActivity,"Full name and User name Required",Toast.LENGTH_SHORT)
                    toast.show()
                }
            }

        }
        buttonLogin.setOnClickListener(clickAction)
    }

    private fun saveLoginState() {
        StoreSession.write(PrefConstant.IS_LOGGED_IN,true)
       // editer = sharedPreferences.edit()
      //  editer.putBoolean(PrefConstant.IS_LOGGED_IN,true)
        //editer.apply()
    }

    private fun saveFullName(fullName: String) {
        StoreSession.write(PrefConstant.FULL_NAME,fullName)
        //editer = sharedPreferences.edit()
       // editer.putString(PrefConstant.FULL_NAME,fullName)
        //editer.apply()
    }
}